### PR TITLE
Feature/tuning pe trans

### PR DESCRIPTION
**Pull request by reyoung**

### CODE REVIEW

File: paddle/fluid/pybind/pybind.cc [hunk 1]
```diff
@@ -510,6 +510,11 @@ All parameter, weight, gradient are variables in Paddle.
              return &self.GetLocalScopes();
            },
            py::return_value_policy::reference)
+      .def("local_scopes_len",
```
Comment (panyx0718) [inline paddle/fluid/pybind/pybind.cc]: Can we avoid exposing these?

**Comment by tonyyang-svail** [comment]
@reyoung Could you please write a few words describing the background of this PR. It is hard to figure it out based on the title and the code.

File: paddle/fluid/pybind/pybind.cc (hunk 1 above)
Comment (reyoung) [reply paddle/fluid/pybind/pybind.cc]: Done. Just add comments to these APIs.

**Comment by reyoung** [comment]
@tonyyang-svail Just add comments for these modifications. There are basically some defects of the previous code.

File: python/paddle/fluid/parallel_executor.py [hunk 2]
```diff
@@ -123,28 +123,65 @@ def __init__(self,
...
-        feed_tensor_dict = {}
-        for i, feed_name in enumerate(feed_dict):
-            feed_tensor = feed_dict[feed_name]
-            if not isinstance(feed_tensor, core.LoDTensor):
-                feed_tensor = core.LoDTensor()
-                feed_tensor.set(feed_dict[feed_name], self._act_places[0])
-            feed_tensor_dict[feed_name] = feed_tensor
+        Args:
+            fetch_list(list): The fetched variable names
+            feed(list|dict|None): The feed variables. If the feed is a dict, tensors in that dict will be splitted
```
Comment (pkuyym) [inline python/paddle/fluid/parallel_executor.py]: Seems exceed 80 columns

**Pull request merged**